File: templates/Uwp/_comp/MVVMLight/Feat.DeepLinking_MenuBar/Activation/SchemeActivationHandler_postaction.cs
//{[{
using Param_RootNamespace.Helpers;
using Param_RootNamespace.Views;
using Param_RootNamespace.ViewModels;
//}]}
namespace Param_RootNamespace.Activation
{
    internal class SchemeActivationHandler : ActivationHandler<ProtocolActivatedEventArgs>
    {
//{[{
        public NavigationServiceEx NavigationService => ViewModelLocator.Current.NavigationService;

        // By default, this handler expects URIs of the format 'wtsapp:sample?paramName1=paramValue1&paramName2=paramValue2'
        protected override async Task HandleInternalAsync(ProtocolActivatedEventArgs args)
        {
            // Create data from activation Uri in ProtocolActivatedEventArgs
            var data = new SchemeActivationData(args.Uri);
            if (data.IsValid)
            {
                MenuNavigationHelper.UpdateView(data.ViewModelName, data.Parameters);
            }
            else if (args.PreviousExecutionState != ApplicationExecutionState.Running)
            {
                // If the app isn't running and not navigating to a specific page based on the URI, navigate to the home page
                NavigationService.Navigate(typeof(ViewModels.Param_HomeNameViewModel).FullName);
            }

            await Task.CompletedTask;
        }

//}]}
    }
}

File: templates/Uwp/_comp/MVVMToolkit/Feat.DeepLinking_MenuBar._VB/Activation/SchemeActivationHandler_postaction.vb
'{[{
Imports Param_RootNamespace.Helpers
Imports Param_RootNamespace.Views
'}]}
Namespace Activation
    Friend Class SchemeActivationHandler
        Inherits ActivationHandler(Of ProtocolActivatedEventArgs)
'{[{

        ' By default, this handler expects URIs of the format 'wtsapp:sample?paramName1=paramValue1&paramName2=paramValue2'
        Protected Overrides Async Function HandleInternalAsync(args As ProtocolActivatedEventArgs) As Task
            ' Create data from activation Uri in ProtocolActivatedEventArgs
            Dim data = New SchemeActivationData(args.Uri)
            If data.IsValid Then
                MenuNavigationHelper.Navigate(data.PageType, data.Parameters)
            End If
            Await Task.CompletedTask
        End Function
'}]}
    End Class
End Namespace
